pkg/aws/policies.go

```diff
@@ -1185,3 +1186,205 @@ func (c *awsClient) GetOpenIDConnectProvider(clusterID string) (string, error) {
 	}
 	return "", nil
 }
+
+func (c *awsClient) IsUpgradedNeededForRole(prefix string, accountID string) (bool, error) {
+	for _, accountRole := range AccountRoles {
+		roleName := fmt.Sprintf("%s-%s-Role", prefix, accountRole.Name)
+		role, err := c.iamClient.GetRole(&iam.GetRoleInput{
+			RoleName: aws.String(roleName),
+		})
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case iam.ErrCodeNoSuchEntityException:
+					return false, errors.NotFound.Errorf("Roles with the prefix'%s' not found", prefix)
+				}
+			}
+			return false, err
+		}
+		isCompatible, err := c.validateRoleUpgradeVersionCompatibility(aws.StringValue(role.Role.RoleName),
+			DefaultPolicyVersion, accountID)
+
+		if err != nil {
+			return false, err
+		}
+		if isCompatible {
```

Comment:
Think we should invert this check, as if the first role checked is valid, then we return here and not check the remaining roles. 

```
if !isCompatible {
   return true, nil
}
```